File: src/01-easy/mime-type/solution.swift
let n = Int(readLine()!)!
let q = Int(readLine()!)!
var m = [String: String]()
for _ in 0..<n {
    let p = readLine()!.split(separator: " ")
    m[String(p[0]).lowercased()] = String(p[1])
}
for _ in 0..<q {
    let f = readLine()!
    if let dot = f.lastIndex(of: ".") {
        let ext = String(f[f.index(after: dot)...]).lowercased()
        print(m[ext] ?? "UNKNOWN")
    } else {
        print("UNKNOWN")
    }
}

File: src/01-easy/mime-type/solution.scala
import scala.io.StdIn
object Solution {
  def main(args: Array[String]): Unit = {
    val n = StdIn.readInt()
    val q = StdIn.readInt()
    val m = (1 to n).map { _ =>
      val p = StdIn.readLine().split(" ")
      p(0).toLowerCase -> p(1)
    }.toMap
    (1 to q).foreach { _ =>
      val f = StdIn.readLine()
      val dot = f.lastIndexOf('.')
      if (dot == -1) println("UNKNOWN")
      else println(m.getOrElse(f.substring(dot + 1).toLowerCase, "UNKNOWN"))
    }
  }
}
